Bump envoy-gloo to support Base64 Inja Transformation Functions (#7255)
solo-io/gloo: docs/content/reference/api/github.com/solo-io/gloo/projects/gloo/api/external/envoy/extensions/transformation/transformation.proto.sk.md

Defines an [Inja template](https://github.com/pantor/inja) that will be
rendered by Gloo. In addition to the core template functions, the Gloo
transformation filter defines the following custom functions:
- header(header_name): returns the value of the header with the given name.
- extraction(extractor_name): returns the value of the extractor with the
given name.
- env(env_var_name): returns the value of the environment variable with the
given name.
- body(): returns the request/response body.
- context(): returns the base JSON context (allowing for example to range on
a JSON body that is an array).

> - request_header(header_name): return the value of the request header with

Reviewer: ```suggestion
- request_header(header_name): returns the value of the request header with
```

Author: Resolved here: https://github.com/solo-io/gloo/pull/7255/commits/3ccb5a463d6c8630dbaa77e569fc6ae9ca9bd59d

> the given name. Useful for including request header values in response

Author: Resolved here: https://github.com/solo-io/gloo/pull/7255/commits/3ccb5a463d6c8630dbaa77e569fc6ae9ca9bd59d

transformations.

> - base64_encode(string): encodes the input string as base64

Reviewer: ```suggestion
- base64_encode(string): encodes the input string to base64.
```

Author: Resolved here: https://github.com/solo-io/gloo/pull/7255/commits/3ccb5a463d6c8630dbaa77e569fc6ae9ca9bd59d

> - base64_decode(string): decodes the input string from base64

Reviewer: ```suggestion
- base64_decode(string): decodes the input string from base64.
```

Author: Resolved here: https://github.com/solo-io/gloo/pull/7255/commits/3ccb5a463d6c8630dbaa77e569fc6ae9ca9bd59d

- substring(string, start_pos, substring_len): returns a substring of the 

> input string, starting at start_pods and extending for substring_len

Reviewer: ```suggestion
input string, starting at `start_pods` and extending for `substring_len`
```

Author: resolved here: https://github.com/solo-io/gloo/pull/7255/commits/3ccb5a463d6c8630dbaa77e569fc6ae9ca9bd59d

> characters. If length is not provided or substring_len is <= 0, the 

Reviewer: ```suggestion
characters. If no length is provided or `substring_len` is <= 0, the 
```

Author: Forwarding the conversation on this point to the thread about the same comment in `projects/gloo/api/external/envoy/extensions/transformation/transformation.proto` -- The changes in this file automatically mirror the changes in that file.

> substring will extend to the end of the string.

Reviewer: ```suggestion
substring extends to the end of the input string.
```

Author: Forwarding the conversation on this point to the thread about the same comment in `projects/gloo/api/external/envoy/extensions/transformation/transformation.proto` -- The changes in this file automatically mirror the changes in that file.
